include/nighthawk/common/poolable.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+
+namespace Nighthawk {
+
+class Poolable {
+public:
+  virtual ~Poolable() = default;
+  virtual void orphan() PURE;
```

Comment:
can you say more about the orphan verb and the orphaned property? I'm not sure exactly what the implication is.

Reply:
Improved naming, added doxygen comments.